src/EditorFeatures/CSharp/BlockCommentEditing/BlockCommentEditingCommandHandler.cs

```diff
@@ -0,0 +1,177 @@
+                }
+                else
+                {
+                    //      /*
+                    //   |   *
+                    return " * ";
+                }
+            }
+
+            return null;
+        }
+
+        private static bool BlockCommentEndsRightAfterCaret(SnapshotPoint caretPosition)
+        {
+            var snapshot = caretPosition.Snapshot;
+            return (caretPosition.Position + 2 > snapshot.Length) ? false : snapshot.GetText(caretPosition, 2) == "*/";
+        }
+
+        private static string GetPaddingOrIndentation(ITextSnapshotLine currentLine, int caretPosition, int firstNonWhitespacePosition, string exteriorText)
+        {
+            Debug.Assert(caretPosition >= firstNonWhitespacePosition + exteriorText.Length);
+
+            var firstNonWhitespaceOffset = firstNonWhitespacePosition - currentLine.Start;
+            Debug.Assert(firstNonWhitespaceOffset > -1);
+
+            var lineText = currentLine.GetText();
+            if ((lineText.Length == firstNonWhitespaceOffset + exteriorText.Length))
+            {
+                //     *|
+                return " ";
+            }
+
+            var interiorText = lineText.Substring(firstNonWhitespaceOffset + exteriorText.Length);
+            var interiorFirstNonWhitespaceOffset = interiorText.GetFirstNonWhitespaceOffset() ?? -1;
+
+            if (interiorFirstNonWhitespaceOffset == 0)
+            {
+                //    /****|
+                return " ";
+            }
+
+            var interiorFirstWhitespacePosition = firstNonWhitespacePosition + exteriorText.Length;
+            if (interiorFirstNonWhitespaceOffset == -1 || caretPosition <= interiorFirstWhitespacePosition + interiorFirstNonWhitespaceOffset)
+            {
+                // *  |
+                // or
+                // *  |  1.
+                //  ^^
+                return currentLine.Snapshot.GetText(interiorFirstWhitespacePosition, caretPosition - interiorFirstWhitespacePosition);
+            }
+            else
+            {
+                // *   1. |
+                //  ^^^
+                return currentLine.Snapshot.GetText(interiorFirstWhitespacePosition, interiorFirstNonWhitespaceOffset);
+            }
+        }
+
+        private static bool IsCaretInsideBlockCommentSyntax(SnapshotPoint caretPosition)
+        {
+            var document = caretPosition.Snapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document == null)
+            {
+                return false;
+            }
+
+            var syntaxTree = document.GetSyntaxTreeAsync().WaitAndGetResult(CancellationToken.None);
+            var trivia = syntaxTree.FindTriviaAndAdjustForEndOfFile(caretPosition, CancellationToken.None);
+
+            return trivia.RawKind == (int)SyntaxKind.MultiLineCommentTrivia || trivia.RawKind == (int)SyntaxKind.MultiLineDocumentationCommentTrivia;
```

Comment:
It's weird that I actually know this method, but could not find it in the completion list, thought I missed a namespace or something. Will change.